src/main/java/com/openshift/internal/restclient/DefaultClient.java

```diff
@@ -145,23 +121,9 @@ public String getResourceURI(IResource resource) {
 	@SuppressWarnings("unchecked")
 	@Override
 	public <T extends IResource> List<T> list(String kind, String namespace, Map<String, String> labels) {
-		try {
-			if(!typeMapper.isSupported(kind))
-				// TODO: replace with specific runtime exception
-				throw new RuntimeException("No OpenShift resource endpoint for type: " + kind);
-			URLBuilder builder = new URLBuilder(this.baseUrl, typeMapper)
-				.kind(kind)
-				.namespace(namespace);
-			final URL endpoint = builder.build();
-			String response = client.get(endpoint,  IHttpClient.DEFAULT_READ_TIMEOUT);
-			LOGGER.debug(String.format("List Response: %s:", response));
-			List<T> items = (List<T>) factory.createList(response, kind);
-			return filterItems(items, labels); //client filter until we can figure out how to restrict with a server call
-		} catch (HttpClientException e){
-			throw createOpenShiftException(String.format("Could not list %s resources in namespace %s: %s", kind, namespace, e.getMessage()), e);
-		} catch (SocketTimeoutException e) {
-			throw new OpenShiftException(e, "Socket timeout listing resources");
-		} 
+		IList list = get(kind, namespace);
```

Comment:
TypeMapper is used by the UrlBuilder and checked for supported types in UrlBuilder#buildWithNamespaceInPath and will throw an UnsupportedEndpointException